addon/components/frost-date-picker.js

```diff
@@ -68,7 +94,8 @@ export default Component.extend(EventsProxyMixin, {
     run.scheduleOnce('sync', this, function () {
       const pikadayOptions = {
         field: this.$('input')[0],
-        onSelect: run.bind(this, this._onSelect)
+        onSelect: run.bind(this, this._onSelect),
```

Comment:
I usually don't like to use `bind` on event handlers without having a way to reference the original handler. Have you verified that `onSelect` get's unregistered.